helpers/import.js

```diff
@@ -38,115 +32,93 @@ exports.importTable = (gtfs, tableName) => {
-
-   decoder will accumulate any "lost" utf8 character at the end of the batch and accumulate it for the next
-   iteration.
-    */
-  const decoder = new StringDecoder('utf8');
-
-  while (position < buffer.length) {
-    rowsSlice = decoder.write(buffer.slice(position, Math.min(buffer.length, position + batchLength)));
+function processGtfsTable(gtfs, fileContent, tableName, indexKeys) {
+  const parsedFileContent = Papa.parse(fileContent, {
+    delimiter: ',',
+    skipEmptyLines: true,
+  });
 
-    if (regexPatternObjects) {
-      regexPatternObjects.forEach(({ regex, pattern }) => {
-        const modifiedRowsSlice = rowsSlice.replace(regex, pattern || '');
-
-        if (modifiedRowsSlice !== rowsSlice) {
-          process.notices.addInfo(__filename, `Applying regex replace to table: "${tableName}". regex: "${regex}".`);
-          rowsSlice = modifiedRowsSlice;
-        }
-      });
-    }
+  if (parsedFileContent.errors.length) {
+    let errorMessage = `Invalid rows in table ${tableName}:\n`;
 
-    rowsSlice.split('\n').forEach((row, i) => {
-      if (i === 0 && merge) {
-        rows[rows.length - 1] += row;
-      } else {
-        rows.push(row);
-      }
+    parsedFileContent.errors.forEach((error) => {
+      errorMessage += `Line: ${error.row}
+Issue: ${error.message}
+Row: ${parsedFileContent.data[error.row].join(',')}`;
     });
 
-    merge = rowsSlice[rowsSlice.length] !== '\n';
-    position += batchLength;
+    if (gtfs._shouldThrow === true) {
+      throw new Error(errorMessage);
+    }
   }
 
-  return rows;
-}
+  const [keys, ...rows] = parsedFileContent.data;
 
-function processRows(gtfs, tableName, indexKeys, rows, shouldThrow) {
-  let table = (indexKeys.setOfItems) ? new Set() : new Map();
+  checkThatKeysIncludeIndexKeys(keys, indexKeys, tableName);
 
-  if (rows === undefined || rows === null || rows.length === 0) {
-    return table;
-  }
+  const trimmedKeys = keys.map(key => key.trim());
+  const GtfsRow = createGtfsClassForKeys(trimmedKeys);
 
-  const sortedKeys = fromCsvStringToArray(rows[0], tableName).map(key => key.trim());
+  return processGtfsTableRows(gtfs, tableName, trimmedKeys, rows, indexKeys, GtfsRow);
+}
 
-  const GtfsRow = createGtfsClassForKeys(sortedKeys);
+function processGtfsTableRows(gtfs, tableName, keys, rows, indexKeys, GtfsRow) {
+  let table = (indexKeys.setOfItems) ? new Set() : new Map();
 
-  checkThatKeysIncludeIndexKeys(sortedKeys, indexKeys, tableName);
+  const regexPatternObjects = gtfs._regexPatternObjectsByTableName.get(tableName);
 
-  eachWithLog(`Importation:${tableName}`, rows, (row, index) => {
-    if (index === 0 || !row || !row.trim) {
-      return;
+  rows.forEach((row) => {
+    if (regexPatternObjects) {
+      row = applyRegexPatternObjectsByTableName(regexPatternObjects, keys, row, tableName);
     }
 
-    row = row.trim();
+    const trimmedRow = row.map(value => value.trim());
+    const gtfsRow = new GtfsRow(trimmedRow);
```

Comment:
You're recreating the GtfsRow for each for, which defeats the memory saving purpose (creating the class n time + object n time, instead of class 1 time + object n time). The keys will be copied the same way than before the optimization.

Reply:
Great point. I will update this